http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -422,6 +428,54 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
       }
     } yield token
 
+  /**
+   * Looks up the workspace parent Sam resource for the given google project.
+   *
+   * This method is used to populate workspaceId for Leonardo resources created
+   * with the v1 routes, which are in terms of google project. Once Leonardo clients
+   * are migrated to the v2 routes this method can be removed.
+   *
+   * Although we expect all google projects to have workspace parents in Sam, this
+   * method will not fail if a workspace cannot be retrieved. Logs and metrics are
+   * emitted for successful and failed workspace retrievals.
+   *
+   * @param userInfo the user info containing an access token
+   * @param googleProject the google project whose workspace parent to look up
+   * @param ev trace id
+   * @return optional workspace ID
+   */
+  override def lookupWorkspaceParentForGoogleProject(userInfo: UserInfo, googleProject: GoogleProject)(implicit
+    ev: Ask[F, TraceId]
+  ): F[Option[WorkspaceId]] = for {
+    traceId <- ev.ask
+
+    // Get resource parent from Sam
+    authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
+    parent <- samDao
+      .getResourceParent(authHeader, ProjectSamResourceId(googleProject))
+
+    // Annotate error cases but don't fail
```

Comment:
We can always do an audit of the prod database for runtimes created after this PR is merged, with workspace ID still being NULL

Reply:
Yup, true